exodus_lambda/functions/origin_response/origin_response.py

```diff
@@ -24,6 +26,22 @@ def handler(self, event, context):
         request = event["Records"][0]["cf"]["request"]
         response = event["Records"][0]["cf"]["response"]
 
+        try:
+            want_digest = request["headers"]["want-digest"][0]["value"]
+        except KeyError:
+            LOG.debug("Could not read want-digest from request", exc_info=True)
+            want_digest = None
+
+        # TODO: Investigate cf-s3 response errors, ensure no digest if error
+        if want_digest:
+            sha256_object_key = basename(request["uri"])
+            response["headers"]["digest"] = [
```

Comment:
> Question: do you also have to check whether the response from origin was successful? If we got an error or other unusual response, I don't think we should add the digest. This is a question since I don't know whether the origin-response event is designed so that the lambda is invoked also on failures (I guess that it probably is, since there's valid use-cases to do things with failures from origin).